certbot/tests/util.py

```diff
@@ -328,22 +329,21 @@ def setUp(self):
 
     def tearDown(self):
         """Execute after test"""
-        # On Windows we have various files which are not correctly closed at the time of tearDown.
-        # For know, we log them until a proper file close handling is written.
-        # Useful for development only, so no warning when we are on a CI process.
-        def onerror_handler(_, path, excinfo):
-            """On error handler"""
-            if not os.environ.get('APPVEYOR'): # pragma: no cover
-                message = ('Following error occurred when deleting the tempdir {0}'
-                           ' for path {1} during tearDown process: {2}'
-                           .format(self.tempdir, path, str(excinfo)))
-                warnings.warn(message)
-        shutil.rmtree(self.tempdir, onerror=onerror_handler)
+        # Cleanup opened resources after a test. This is usually done through atexit handlers in
+        # Certbot, but during pytest, atexit will kicks in only after each tearDown execution.
+        # It is a problem on Windows, that does not accept to clean resources before closing them.
+        logging.shutdown()
+        util._release_locks()  # pylint: disable=protected-access
 
-class ConfigTestCase(TempDirTestCase):
-    """Test class which sets up a NamespaceConfig object.
+        def handle_rw_files(_, path, __):
+            """Handle read-only files, that will fail to be removed on Windows."""
```

Comment:
This is probably fine, by where are these read-only files being created?

Reply:
Typically ACME account keys, that have `0o400`. If this is fine in Linux when you are root, it is not on Windows. From TempDirTestCase point of view, whose objective is to delete the temporary directory and all its content to matter what, if think it is a sane approach to put the abstraction here.

Reply:
Thanks for the explanation!